### PR TITLE
Parse email files time out

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/43897

## Description
Fixed an issue with attachments with unknown encoding

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_6_5.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### ParseEmailFiles
+- Fixed an issue where the attachment name was in an unknown encoding.
```
Comment (reutshal) [inline Packs/CommonScripts/ReleaseNotes/1_6_5.md:4]:
```suggestion
- Fixed an issue where the encoding of an attachment name was mishandled
```

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py [hunk 2]
```diff
@@ -3433,7 +3433,8 @@ def convert_to_unicode(s, is_msg_header=True):
                     res += replace_decoded
                 ENCODINGS_TYPES.add(encoding)
             else:
-                res += decoded_s
+                demisto.debug('Could not find the encoding type of the string, Decoded by default with utf-8 ')
```
Comment (reutshal) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py:3436]:
```suggestion
                demisto.debug('Could not find the encoding type of the string, decoding by default with utf-8')
```

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles_test.py [hunk 3]
```diff
@@ -1089,6 +1089,27 @@ def test_eml_contains_htm_attachment_empty_file(mocker):
...
+    """
+    Given: An email containing an attachment with unknown encoded name.
+    When: Parsing a valid email file with default parameters.
+    Then: The file name is parsed as expected.
+    """
+    mocker.patch.object(demisto, 'args', return_value={'entryid': 'test'})
+    mocker.patch.object(demisto, 'executeCommand',
+                        side_effect=exec_command_for_file('Unknown_encode_attachment_name.eml'))
+    mocker.patch.object(demisto, 'results')
+    # validate our mocks are good
```
Comment (reutshal) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles_test.py:1102]: Why do we need this?

**Review by reutshal — APPROVED** [review]
Nice work

File: Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles_test.py (hunk 3 above)
Comment (MosheEichler) [reply Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles_test.py:1102]: This is actually unnecceary (I just copied this section of code from another UT) I deleted it

File: Packs/CommonScripts/Scripts/ParseEmailFiles/test_data/Unknown_encode_attachment_name.eml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+Content-Type: multipart/alternative;
+ boundary="_111111_"
+
+--_123456_
+Content-Type: application/octet-stream;
+	name="04. ���A������ИIƷ�|���υf��.pdf"
+Content-Transfer-Encoding: base64
+Content-Disposition: attachment;
+	filename="04. ���A������ИIƷ�|���υf��.pdf"
+
```
Comment (reutshal) [inline Packs/CommonScripts/Scripts/ParseEmailFiles/test_data/Unknown_encode_attachment_name.eml:19]: Let's relace the content of this file with some example content. We dont know if there is sensitive data there.
Also, is the name of the file the one given to us by the customer?
If so, please ask for their permission to use this in our testing.
Content is an open-source repo and we dont want any sensitive information to be there,

File: Packs/CommonScripts/ReleaseNotes/1_6_6.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### ParseEmailFiles
+- Fixed an issue where the encoding of an attachment name was mishandled
```
Comment (amshamah419) [inline Packs/CommonScripts/ReleaseNotes/1_6_6.md:4]:
```suggestion
- Fixed an issue where the encoding of an attachment name was not handled properly.
```

**Review by reutshal — APPROVED** [review]

**Review by reutshal — APPROVED** [review]

**Comment by moishce** [comment]
@MosheEichler Please add the changes to https://github.com/demisto/parse-emails as well

**Comment by MosheEichler** [comment]
@moishce Please take a look
https://github.com/demisto/parse-emails/pull/21

**Pull request merged**